hooks/useContract.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+import { Address, PublicClient, getContract } from 'viem';
+
+export function useContract<TAbi extends readonly unknown[]>(address: Address, abi: TAbi, publicClient: PublicClient) {
+  const [contract, setContract] = useState(() =>
```

Comment:
Why do you need to contract in a state? Have you considered just memoizing it with `useMemo`?

Reply:
I'll use a `useMemo` as It will create the contract only when its dependencies change, which is more efficient than using state. Good catch!

Reply:
got it